docs/source/cheat_sheet_py3.rst

```diff
@@ -0,0 +1,280 @@
+       def my_class_method(self, num: int, str1: str) -> str:
+           return num * str1
+
+
+
+   # User-defined classes are written with just their own names.
+   x = MyClass() # type: MyClass
+
+
+Other stuff
+***********
+
+.. code-block:: python
+
+   import sys
+   import re
+   # typing.Match describes regex matches from the re module.
+   from typing import Match, AnyStr, IO
+   x = re.match(r'[0-9]+', "15") # type: Match[str]
+
+   # You can use AnyStr to indicate that any string type will work
+   # but not to mix types
+   def full_name(first: AnyStr, last: AnyStr) -> AnyStr:
+       return first+last
+   full_name('Jon','Doe') # same str ok
+   full_name(b'Bill', b'Bit') # same binary ok
+   full_name(b'Terry', 'Trouble') # different str types, fails
+
+   # Use IO[] for functions that should accept or return any
+   # object that comes from an open() call. The IO[] does not
+   # distinguish between reading, writing or other modes.
+   def get_sys_IO(mode='w') -> IO[str]:
+       if mode == 'w':
+           return sys.stdout
+       elif mode == 'r':
+           return sys.stdin
+       else:
+           return sys.stdout
+
+   # forward references are useful if you want to referemce a class before it is designed
+   
+   def(foo: A) -> int: # this will fail
+       ...
+   
+   class A:
+       pass
+       
+   # however, using the string 'A', it will pass as long as there is a class of that name later on
+   def(foo: 'A') -> int:
+       ...
+
+   # TODO: add TypeVar and a simple generic function
+
+Variable Annotation in Python 3.6 with PEP 526
+**********************************************
+
+Python 3.6 brings new syntax for annotating variables with `PEP 526 <https://www.python.org/dev/peps/pep-0526/>`_.
+
+
+
+.. code-block:: python
+
+   from typing import ClassVar
+   # annotation is similar to arguments to functions
+   name: str = "Eric Idle"
+   
+   # you can annotate objects in more complex expresions such as
+   a = {}
+   a['b']: int # the value is checked to be of type int
+   
+   b = Car() : Car
+   b.doors: int = 4 # you can type annotate instance variables after class instantiation
+   
+   # tuple unpacking can be done as follows
+   tu: Tuple[str, ...] = ('a', 'b', 'c')
+   
+   # annotations are not checked at runtime
+   year: int = '1972' # error in type checking, but work at runtime
+   
+   # these are all equivalent
+   hour = 24 # type: int
+   hour: int; hour = 24
+   hour: int = 24
+   
+   # you do not (!) need to initialize a variable to annotate it
+   a: int # ok for type checking and runtime
+   # which is useful in conditional branches
+   child: bool
+   if age < 18:
+       child = True
+   else:
+       child = False
+   
+   # annotations for classes are for instance variables (those created in __init__ or __new__)
+   class Battery:
+       charge_percent: int = 100 # this is an instance variable with a default value
+       capacity: int # an instance variable without a default
```

Comment:
Yeah, an example like the above (well hopefully not using a class named `A` :-P) would be nice.